### PR TITLE
Lots of different cleanups

### PR DESCRIPTION
**Pull request by kevmoo**
Note we still have 3 failures – something is broken in 1.24

### CODE REVIEW

**Comment by kevmoo** [comment]
> Note we still have 3 failures – something is broken in 1.24

Or maybe not. Anyway...

File: test/runner/compact_reporter_test.dart [hunk 1]
```diff
@@ -490,6 +490,12 @@ $tests
       return line.substring(indentation);
     }).join("\n");
 
-    expect(actual, equals(expected));
+    // In Dart 1.24, stack traces with Future constructors output as
+    // `new Future` instead of `Future.Future`.
+    // Support running tests in both old and new styles.
```
Comment (natebosch) [inline test/runner/compact_reporter_test.dart]: how long do we expect we'll keep backwards compatibility?

Comment (kevmoo) [inline test/runner/compact_reporter_test.dart]: At least until 1.24 goes gold – and hopefully a bit longer. A release or two?

**Comment by natebosch** [comment]
Did you intend to keep using the same branch/pull request for the other changes?

**Pull request merged**